client/src/annoMatrix/views.ts

```diff
@@ -4,37 +4,59 @@
 Views on the annomatrix.  all API here is defined in viewCreators.js and annoMatrix.js.
 */
 import clip from "../util/clip";
-import AnnoMatrix from "./annoMatrix";
-import { _whereCacheCreate } from "./whereCache";
+import AnnoMatrix, { ObsColumnValue } from "./annoMatrix";
+import { _whereCacheCreate, WhereCache } from "./whereCache";
 import { _isContinuousType, _getColumnSchema } from "./schema";
+import {
+  Dataframe,
+  DenseInt32Index,
+  IdentityInt32Index,
+  KeyIndex,
+} from "../util/dataframe";
+import { Query } from "./query";
+import {
+  AnnotationColumn,
+  ColumnSchema,
+  Field,
+  LayoutColumn,
+} from "../common/types/entities";
 
-class AnnoMatrixView extends AnnoMatrix {
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-  constructor(viewOf: any, rowIndex = null) {
-    // @ts-expect-error ts-migrate(2531) FIXME: Object is possibly 'null'.
+type MapFn = (
+  field: Field,
+  colLabel: string,
+  colSchema: ColumnSchema,
+  colData: string[],
+  df: Dataframe
+) => string[];
+
+abstract class AnnoMatrixView extends AnnoMatrix {
+  constructor(
+    viewOf: AnnoMatrix,
+    rowIndex: DenseInt32Index | IdentityInt32Index | KeyIndex | null = null
```

Comment:
I will most likely add a `RowLabel` and `ColumnLabel` type to the util/dataframe for cases like this.

Reply:
Dataframe now has:
* LabelType - for a singleton label
* LabelIndex - a row/col label index  (ie, L35)